src/privatesend-server.cpp

```diff
@@ -772,12 +772,17 @@ bool CPrivateSendServer::AddUserToExistingSession(const CPrivateSendAccept& dsa,
     nTimeLastSuccessfulStep = GetTime();
     vecSessionCollaterals.push_back(MakeTransactionRef(dsa.txCollateral));
 
-    LogPrintf("CPrivateSendServer::AddUserToExistingSession -- new user accepted, nSessionID: %d  nSessionDenom: %d (%s)  vecSessionCollaterals.size(): %d\n",
-        nSessionID, nSessionDenom, CPrivateSend::GetDenominationsToString(nSessionDenom), vecSessionCollaterals.size());
+    LogPrintf("CPrivateSendServer::AddUserToExistingSession -- new user accepted, nSessionID: %d  nSessionDenom: %d (%s)  vecSessionCollaterals.size(): %d  nSessionMaxParticipants: %d\n",
+        nSessionID, nSessionDenom, CPrivateSend::GetDenominationsToString(nSessionDenom), vecSessionCollaterals.size(), nSessionMaxParticipants);
 
     return true;
 }
 
+bool CPrivateSendServer::IsSessionReady()
+{
+    return nSessionMaxParticipants != 0 && (int)vecSessionCollaterals.size() >= nSessionMaxParticipants;
```

Comment:
I thought about this change, and I agree with it, but didn't see any real necessity to based on when IsSessionReady is called

Reply:
The issue is that mixing never starts because the first `DSACCEPT` will never pass here https://github.com/dashpay/dash/blob/develop/src/privatesend-server.cpp#L37 (`vecSessionCollaterals.size()` is equal to `nSessionMaxParticipants` because they both are equal to zero which means `IsSessionReady()` returns `true`) and thus new session won't be created here https://github.com/dashpay/dash/blob/develop/src/privatesend-server.cpp#L81